### PR TITLE
Adding new Swagger nuget package for creating swagger doc and ui

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: samples/SampleApp/App_Start/Startup.Swagger.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        public static void ConfigureSwagger(HttpConfiguration config)
+        {
+            config.Services.Replace(typeof(IApiExplorer), new MobileAppApiExplorer(config));
+            config
+               .EnableSwagger(c =>
+               {
+                   c.SingleApiVersion("v1", "brettsam1201Service");
+                   c.OperationFilter<MobileAppHeaderFilter>();
+                   c.SchemaFilter<MobileAppSchemaFilter>();
+                   c.AppServiceAuthentication("https://brettsam1201.azurewebsites.net/", "aad");
```
Comment (aziel) [inline samples/SampleApp/App_Start/Startup.Swagger.cs]: Do you intend to check these in?

File: src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/o2c.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿<script>
+    var tokenAsJson = null;
+    var qp = {};
+    var i = location.href.indexOf('#token=');
+    if (i > 0) {
+        tokenAsJson = decodeURIComponent(location.href.substring(i + 7));
```
Comment (aziel) [inline src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/o2c.html]: what is the "+7" from?

File: src/Microsoft.Azure.Mobile.Server.Swagger/packages.config [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net45" />
+  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net45" />
+  <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net45" />
+  <package id="Microsoft.Bcl.Build" version="1.0.21" targetFramework="net45" />
+  <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net45" />
+  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net45" />
+  <package id="Swashbuckle.Core" version="5.2.2" targetFramework="net452" />
```
Comment (aziel) [inline src/Microsoft.Azure.Mobile.Server.Swagger/packages.config]: should all these be net 452 now?

File: src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/swagger-oauth.js [hunk 4]
```diff
@@ -0,0 +1,290 @@
+﻿var appName;
```
Comment (aziel) [inline src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/swagger-oauth.js]: We're these auto-generated by the Swagger nuget or copy-pasted from elsewhere?

File: samples/SampleApp/App_Start/Startup.Swagger.cs (hunk 1 above)
Comment (brettsam) [inline samples/SampleApp/App_Start/Startup.Swagger.cs]: Nope!

File: src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/o2c.html (hunk 2 above)
Comment (brettsam) [inline src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/o2c.html]: It's offsetting from '#token=', which is 7 characters. This is lifted from our javascript sdk: https://github.com/Azure/azure-mobile-services/blob/8b176174c5d52c4ecd45450c89e91f06a423bdd6/sdk/Javascript/src/LoginUis/WebAuthBroker.js#L39

File: src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/swagger-oauth.js (hunk 4 above)
Comment (brettsam) [inline src/Microsoft.Azure.Mobile.Server.Swagger/Swagger/swagger-oauth.js]: This file is taken from Swashbuckle and tweaked to work with App Service Auth (our parameter names are different). I'll attribute it and call out exactly what I changed.

File: samples/SampleApp/App_Start/Startup.Swagger.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            // Use the custom ApiExplorer that applies constraints. This prevents
+            // duplicate routes on /api and /tables from showing in the Swagger doc.
+            config.Services.Replace(typeof(IApiExplorer), new MobileAppApiExplorer(config));
+            config
+               .EnableSwagger(c =>
+               {
+                   c.SingleApiVersion("v1", "myService");
+
+                   // Tells the Swagger doc that any MobileAppController needs a
+                   // ZUMO -API-VERSION header with default 2.0.0
```
Comment (aziel) [inline samples/SampleApp/App_Start/Startup.Swagger.cs]: ZUMO-API-VERSION

File: samples/SampleApp/SampleApp.csproj [hunk 6]
```diff
@@ -23,6 +23,8 @@
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
     <UseGlobalApplicationHostFile />
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
```
Comment (aziel) [inline samples/SampleApp/SampleApp.csproj]: ?  Empty section.

File: samples/SampleApp/App_Start/Startup.Swagger.cs (hunk 5 above)
Comment (brettsam) [inline samples/SampleApp/App_Start/Startup.Swagger.cs]: Fixed.

File: samples/SampleApp/SampleApp.csproj (hunk 6 above)
Comment (brettsam) [inline samples/SampleApp/SampleApp.csproj]: I don't think this is needed; i'll remove it: http://stackoverflow.com/questions/24049961/what-is-nugetpackageimportstamp-for

File: src/Microsoft.Azure.Mobile.Server.Swagger/packages.config (hunk 3 above)
Comment (brettsam) [inline src/Microsoft.Azure.Mobile.Server.Swagger/packages.config]: I think they should actually all be net45

File: test/Microsoft.Azure.Mobile.Server.Swagger.Test/packages.config [hunk 7]
```diff
@@ -0,0 +1,21 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net452" />
+  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net452" />
```
Comment (aziel) [inline test/Microsoft.Azure.Mobile.Server.Swagger.Test/packages.config]: Then these should probably be "net45" also?

**Comment by aziel** [comment]
Looks good.  Signed off.

**Pull request merged**